### PR TITLE
feat: add SequencerInbox setters

### PR DESCRIPTION
**Pull request by chrstph-dvx**
Part of FS-535

Getters:
https://github.com/OffchainLabs/arbitrum-orbit-sdk/pull/144
https://github.com/OffchainLabs/arbitrum-orbit-sdk/pull/145
https://github.com/OffchainLabs/arbitrum-orbit-sdk/pull/146
https://github.com/OffchainLabs/arbitrum-orbit-sdk/pull/147

Setters:
https://github.com/OffchainLabs/arbitrum-orbit-sdk/pull/148
https://github.com/OffchainLabs/arbitrum-orbit-sdk/pull/149
https://github.com/OffchainLabs/arbitrum-orbit-sdk/pull/150

### CODE REVIEW

File: src/withUpgradeExecutor.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    to: Address;
+    upgradeExecutor: false | Address;
+  },
+) {
+  const { upgradeExecutor } = params;
+  if (!upgradeExecutor) {
+    return {
+      to: params.to,
+      data: encodeFunctionData(params),
+      value: BigInt(0),
```
Comment (spsjvc) [inline src/withUpgradeExecutor.ts:42]:
```suggestion
      value: params.value,
```

File: src/withUpgradeExecutor.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  return {
+    to: upgradeExecutor,
+    data: upgradeExecutorEncodeFunctionData({
+      functionName: 'executeCall',
+      args: [
+        params.to, // target
+        encodeFunctionData(params), // targetCallData
+      ],
+    }),
+    value: BigInt(0),
```
Comment (spsjvc) [inline src/withUpgradeExecutor.ts:55]:
```suggestion
    value: params.value,
```

File: src/withUpgradeExecutor.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      to: params.to,
+      data: encodeFunctionData(params),
+      value: BigInt(0),
+    };
+  }
+
+  return {
+    to: upgradeExecutor,
+    data: upgradeExecutorEncodeFunctionData({
+      functionName: 'executeCall',
```
Comment (spsjvc) [inline src/withUpgradeExecutor.ts:49]: let's also make this configurable but default to `executeCall`

File: src/withUpgradeExecutor.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  args,
+}: EncodeFunctionDataParameters<TAbi, TFunctionName>) {
+  return viemEncodeFunctionData({
+    abi,
+    functionName,
+    args,
+  } as unknown as ViemEncodeFunctionDataParameters<TAbi, TFunctionName>);
+}
+
+export function withUpgradeExecutor<TAbi extends ABIs, TFunctionName extends FunctionName<TAbi>>(
```
Comment (spsjvc) [inline src/withUpgradeExecutor.ts:31]:
```suggestion
export function prepareUpgradeExecutorCallParameters<TAbi extends ABIs, TFunctionName extends FunctionName<TAbi>>(
```

File: src/actions/buildSetIsbatchPoster.ts
[the remark this answers is not in the document]
Comment (chrstph-dvx) [reply src/actions/buildSetIsbatchPoster.ts:1]: It's a function we're calling, not really a factory or a class. Also we have all getters with lowercase name

File: src/withUpgradeExecutor.ts (hunk 3 above)
Comment (chrstph-dvx) [reply src/withUpgradeExecutor.ts:49]: Changed (but limited to `execute` or `executeCall`)

**Review by spsjvc — APPROVED** [review]

**Pull request merged**